src/ir/branch-utils.h

```diff
@@ -28,111 +28,108 @@ namespace BranchUtils {
 // Some branches are obviously not actually reachable (e.g. (br $out
 // (unreachable)))
 
-inline bool isBranchReachable(Break* br) {
-  return !(br->value && br->value->type == Type::unreachable) &&
-         !(br->condition && br->condition->type == Type::unreachable);
+inline bool isBranchReachable(Expression* expr) {
+  // If any child is unreachable, the branch is not taken. Note that expr itself
+  // may be unreachable regardless (as in the case of a simple Break with no
+  // condition, which is still taken).
+  for (auto child : ChildIterator(expr)) {
+    if (child->type == Type::unreachable) {
+      return false;
+    }
+  }
+  return true;
 }
 
-inline bool isBranchReachable(Switch* sw) {
-  return !(sw->value && sw->value->type == Type::unreachable) &&
-         sw->condition->type != Type::unreachable;
+// Perform a generic operation on uses of scope names (branch targets) in an
+// expression. The provided function receives a Name& which it can modify if it
+// needs to.
+inline void operateOnScopeNameUses(Expression* expr,
+                                   std::function<void(Name&)> func) {
```

Comment:
It might be good to templatize this over some callable `F` rather than using `std::function` to avoid the lambdas being converted.

Reply:
I guess that's shorter too, sounds good.

Is there an overhead to the conversion, though? I'd hope `std::function` is a small wrapper around a function pointer?

Reply:
I could imagine that the conversion could defeat the optimizer and prevent the lambda from being inlined, but perhaps the optimizer is smart enough to work around that 🤷‍♂️